types/recharts/index.d.ts

```diff
@@ -730,6 +730,7 @@ export interface ReferenceLineProps extends Partial<PresentationAttributes<numbe
     ifOverflow?: IfOverflowType;
     x?: number | string;
     y?: number | string;
+    segment?: Array<{x: number | string, y: number | string}>;
```

Comment:
Could you please use `ReadonlyArray` instead? That way it would accept both mutable and immutable arrays.

```suggestion
    segment?: ReadonlyArray<{x: number | string, y: number | string}>;
```

Or, even better, you could use tuple. Because despite original types in Recharts repo it expects `segment` to contain [exactly two elements](https://github.com/recharts/recharts/blob/master/src/cartesian/ReferenceLine.tsx#L135).

```ts
interface SegmentItem {
  x: number | string
  y: number | string
}
```

```suggestion
    segment?: readonly [SegmentItem, SegmentItem];
```

Reply:
Thanks for a quick review! The tuple was a way i originaly used when testing but
1) I was unsure about debating with original recharts definition. Although that must not be a problem.
2) Readonly tuples are only supported since typescript 3.4 yet now d.ts file contains requirement only for ts 2.8. That might be a problem. Would it be ok to update the required version?

Reply:
Good point, I'm not sure about updating typescript version so in that case `ReadonlyArray` would be sufficient.

Reply:
I think we are already using tuples in the domain definition, or is it something different?

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/18e6bdc0000d2090029653722a2f90350f1b349b/types/recharts/index.d.ts#L1018

```ts
segment?: Readonly<[SegmentItem, SegmentItem]>;
```

Reply:
You seem to be correct. <T> syntax is fine. Thanks!